Source/WebCore/rendering/TextBoxPainter.cpp

```diff
@@ -366,9 +366,25 @@ void TextBoxPainter<TextBoxPath>::paintForegroundAndDecorations()
             }
         }
     } else {
+        // Check if markedTexts vector contain highlights.
+        bool containsHighlights = markedTexts.containsIf([](const auto& item) {
+            return item.type == MarkedText::Type::Highlight;
+        });
+        if (containsHighlights)
+            StyledMarkedText::removeOverlappingForegroundRanges(styledMarkedTexts);
+
         // Coalesce styles of adjacent marked texts to minimize the number of drawing commands.
         auto coalescedStyledMarkedTexts = StyledMarkedText::coalesceAdjacentWithEqualForeground(styledMarkedTexts);
 
+        if (coalescedStyledMarkedTexts.isEmpty())
+            return;
+
+        // DEBUG JESS
```

Comment:
I guess you forgot to remove this :)

Reply:
Added that debugging code bc a test only fails on linux rn :(